forge/forge/models/config.py

```diff
@@ -20,16 +23,27 @@ def UserConfigurable(
     **kwargs,
 ) -> T:
     # TODO: use this to auto-generate docs for the application configuration
-    return Field(
+    field_info: FieldInfo = Field(
         default,
         *args,
         default_factory=default_factory,
-        from_env=from_env,
         description=description,
         exclude=exclude,
         **kwargs,
-        user_configurable=True,
     )
+    field_info.metadata.append("user_configurable")
+    field_info.metadata.append(("from_env", from_env))
```

Comment:
There's no more `extra` and custom values cannot be passed directly to `Field`, so I used `metadata` instead; https://github.com/pydantic/pydantic/discussions/2419#discussioncomment-7228409